### ISSUE TITLE
Room doesnt contain any smart valves, maybe an error/corruption

### ISSUE THREAD

**Issue by jchasey** [asantaga/wiserHomeAssistantPlatform#4]
Just upgraded to the latest set of code and got some new errors:

Room doesnt contain any smart valves, maybe an error/corruption
6:34 AM custom_components/wiserAPI/wiserHub.py (WARNING)

Get 4 of these in sequence - I have 3 trvs and the main sensor - probably not a coincidence.

**Comment by asantaga** [comment]
I'll check this out.. This is my code which ensures we dont get "None" exceptions.. 

Do u have any rooms without TRVs or Devices?
Apart from the error, which maybe by red herring, does everything else work correctly?

**Comment by jchasey** [comment]
No all rooms that are defined in the app have either a TRV or a thermostat display. 

Other than that all seems ok, although have had all the rooms saying that they are at -20. When the next heating period kicked in, they changed to correct values and are okay now the heating schedule has turned off again.
I think HA was rebooted in the period when the heating was off so possibly a situation where they were not updating correctly?

**Comment by asantaga** [comment]
Strange.. 

Oh for the -20c value, I added a new property to the configuration.yaml file (minimum) which you can set to be the minimum value.. I have it default to -5 as its VERY UNLIKELY the room temperature will ever get to this level.. it stops the charts from showing duff data.. 

I'll check my running server tonight and see if I can see anything there..  

Do you mind sending me a JSON payload from you system? use CURL or postman etc, be good to compare. I know @hueytech had a similar issue which is why I coded up some None checks...

I need to check to see if python has None safe operators like groovy does (ie  hub?.getSomething, where the ?. checks that that all the operators return a value.. RTFM time for me

**Comment by jchasey** [comment]
Sure, could you confirm the format of the command required to send to the wiser hub? Tried using https://github.com/micha/resty but getting a unauthorized error being kicked back, so presume I am formatting the command incorrectly.
Using:
`resty http://10.10.10.11/data/domain/ -H "Accept: application/json"
`GET '{"SECRET":"wiser secret goes here"}'

**Comment by asantaga** [comment]
Are you using **Windows?**
if so then 
1. Enter Windows Powershell
2. type `$headers= @{ 'secret' = '<SECRETHERE'}`
3. type `Invoke-RestMethod -Uri http://<YOUR_HOSTNAME>/data/v2/domain/ -Method Get -Headers $headers  | ConvertTo-Json > output.json`

Obviously replacing SECRETHERE and YOUR_HOSTNAME accordingly

This will produce a file output.json which you can send me

If its **Linux** then its a lot simpler :-)

```
curl -X GET \
  http://SECRETHERE/data/v2/domain/   -H 'SECRET: <SECRETHERE >' 
  -H 'cache-control: no-cache > output.json'
```

**Comment by asantaga** [comment]
Files received via email

**Comment by hueytech** [comment]
FWIW I’m getting the same errors.

**Comment by asantaga** [comment]
Ive checked my install and all looks good here, no errors.. I was working on a dev branch (investigateFixSyncIssues) so what Ive done is merged that to the master..

try pulling the latest and greatest.. .. meanwhilst I'll do more debugging with johns files

**Comment by hueytech** [comment]
Yes it still works but will pull the master again.

**Comment by asantaga** [comment]
.. i think ive found it..

**Comment by asantaga** [comment]
John , your error message was fixed yesterday when hueytech c mentioned the same errors
The only bug I fixed today is where we're getting "Room doesnt contain any smart valves, maybe an error/corruption?" eronously.. Damm Python Tabs!!!

I no longer see stack traces.. however hueytech c will see the warnings because he has rooms setup without any devices in them...

**Comment by asantaga** [comment]
hello all? now working?

**Comment by hueytech** [comment]
Good here thanks. 👍🏻

**Comment by asantaga** [comment]
ok closing , if the errors re-appear then we can reopen

**Comment by mrvladis** [comment]
Hi @asantaga ,

Tried the latest release today and got the same error:
2019-12-28 21:26:36 WARNING (SyncWorker_13) [wiserHeatingAPI.wiserHub]  Room doesnt contain any smart valves, maybe an error/corruption??

**HA**
**System Health**
arch | x86_64
-- | --
dev | false
docker | true
hassio | false
os_name | Linux
python_version | 3.7.4
timezone | UTC
version | 0.102.2
virtualenv | false

**Wiser Room Info:**
    "Room": [
        {
            "id": 9,
            "ManualSetPoint": 220,
            "OverrideType": "EcoIQ",
            "OverrideSetpoint": 220,
            "ScheduleId": 1009,
            "ComfortModeScore": 4256,
            "HeatingRate": 1658,
            "RoomStatId": 53938,
            "SmartValveIds": [
                53937
            ],
            "UfhRelayIds": [],
            "Name": "First Floor Bedroom",
            "Mode": "Auto",
            "DemandType": "Modulating",
            "WindowDetectionActive": false,
            "ControlSequenceOfOperation": "HeatingOnly",
            "HeatingType": "HydronicRadiator",
            "CalculatedTemperature": 22.1,
            "CurrentSetPoint": 22,
            "PercentageDemand": 47,
            "ControlOutputState": "Off",
            "WindowState": "Closed",
            "SetpointOrigin": "FromEcoIQ",
            "DisplayedSetPoint": 22.5,
            "ScheduledSetPoint": 22.5,
            "AwayModeSuppressed": false,
            "RoundedAlexaTemperature": 220,
            "EffectiveMode": "Auto",
            "PercentageDemandForItrv": 47
        },
        {
            "id": 11,
            "ScheduleId": 1011,
            "ComfortModeScore": 6409,
            "HeatingRate": 3134,
            "RoomStatId": 53931,
            "SmartValveIds": [
                53940,
                53934,
                53933
            ],
            "UfhRelayIds": [],
            "Name": "Ground Floor Living Room",
            "Mode": "Auto",
            "DemandType": "Modulating",
            "WindowDetectionActive": false,
            "ControlSequenceOfOperation": "HeatingOnly",
            "HeatingType": "HydronicRadiator",
            "CalculatedTemperature": 23.2,
            "CurrentSetPoint": 22.5,
            "PercentageDemand": 0,
            "ControlOutputState": "Off",
            "WindowState": "Closed",
            "SetpointOrigin": "FromSchedule",
            "DisplayedSetPoint": 22.5,
            "ScheduledSetPoint": 22.5,
            "AwayModeSuppressed": false,
            "RoundedAlexaTemperature": 230,
            "EffectiveMode": "Auto",
            "PercentageDemandForItrv": 0
        },
        {
            "id": 13,
            "ScheduleId": 1013,
            "ComfortModeScore": 58940,
            "HeatingRate": 7200,
            "RoomStatId": 53939,
            "SmartValveIds": [
                53935
            ],
            "UfhRelayIds": [],
            "Name": "First Floor Guest Room",
            "Mode": "Auto",
            "DemandType": "Modulating",
            "WindowDetectionActive": false,
            "ControlSequenceOfOperation": "HeatingOnly",
            "HeatingType": "HydronicRadiator",
            "CalculatedTemperature": 21.5,
            "CurrentSetPoint": 22.5,
            "PercentageDemand": 100,
            "ControlOutputState": "On",
            "WindowState": "Closed",
            "SetpointOrigin": "FromSchedule",
            "DisplayedSetPoint": 22.5,
            "ScheduledSetPoint": 22.5,
            "RoundedAlexaTemperature": 215,
            "EffectiveMode": "Auto",
            "PercentageDemandForItrv": 100
        },
        {
            "id": 14,
            "ScheduleId": 1014,
            "ComfortModeScore": 870,
            "HeatingRate": 1175,
            "SmartValveIds": [
                53936
            ],
            "UfhRelayIds": [],
            "Name": "First Floor Wardrobe Room",
            "Mode": "Auto",
            "DemandType": "Modulating",
            "WindowDetectionActive": false,
            "ControlSequenceOfOperation": "HeatingOnly",
            "HeatingType": "HydronicRadiator",
            "CalculatedTemperature": 22.4,
            "CurrentSetPoint": 22,
            "PercentageDemand": 0,
            "ControlOutputState": "Off",
            "WindowState": "Closed",
            "SetpointOrigin": "FromSchedule",
            "DisplayedSetPoint": 22,
            "ScheduledSetPoint": 22,
            "RoundedAlexaTemperature": 225,
            "EffectiveMode": "Auto",
            "PercentageDemandForItrv": 0
        },
        {
            "id": 15,
            "ScheduleId": 1015,
            "ComfortModeScore": 41411,
            "HeatingRate": 6451,
            "RoomStatId": 1368,
            "UfhRelayIds": [],
            "Name": "Ground Floor Kitchen",
            "Mode": "Auto",
            "DemandType": "Modulating",
            "WindowDetectionActive": false,
            "ControlSequenceOfOperation": "HeatingOnly",
            "HeatingType": "HydronicRadiator",
            "CalculatedTemperature": 23.2,
            "CurrentSetPoint": 22,
            "PercentageDemand": 0,
            "ControlOutputState": "Off",
            "SetpointOrigin": "FromSchedule",
            "DisplayedSetPoint": 22,
            "ScheduledSetPoint": 22,
            "AwayModeSuppressed": false,
            "RoundedAlexaTemperature": 230,
            "EffectiveMode": "Auto",
            "PercentageDemandForItrv": 0
        }
    ],

Installation was performed via HASC.

Any thoughts?

**Comment by asantaga** [comment]
this is because you have rooms without any TRVs.. I think its  the same as https://github.com/asantaga/wiserHomeAssistantPlatform/issues/28..

I really need to fix this, many people seam to have this scenario

**Comment by mrvladis** [comment]
Indeed I do. Kitchen is on its own channel, so no need for TRVs.

**Comment by mrvladis** [comment]
Should we may be have this re-opened?

**Comment by asantaga** [comment]
Hey @mrvladis , lets leave this one closed and reopen issue #28 if the issue persists on the latest release

**Comment by mrvladis** [comment]
@asantaga Sure, it is working well for me.
Sorry, I thought I've updated the thread.
